gcp_variant_transforms/beam_io/vcf_file_size_io.py

```diff
@@ -0,0 +1,225 @@
+# Copyright 2018 Google Inc.  All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A source for estimating the size of VCF files when processed by vcf_to_bq."""
+
+from __future__ import absolute_import
+
+from typing import Iterable, List, Tuple  # pylint: disable=unused-import
+import logging
+
+import apache_beam as beam
+from apache_beam import coders
+from apache_beam import transforms
+from apache_beam.io import filebasedsource
+from apache_beam.io import range_trackers  # pylint: disable=unused-import
+from apache_beam.io import filesystem
+from apache_beam.io import filesystems
+from apache_beam.io import iobase
+
+from gcp_variant_transforms.beam_io import vcfio
+
+
+def _get_file_sizes(file_pattern):
+  # type: (str) -> List[FileSizeInfo]
+  file_sizes = []
+  match_result = filesystems.FileSystems.match([file_pattern])[0]
+  for file_metadata in match_result.metadata_list:
+    compression_type = filesystem.CompressionTypes.detect_compression_type(
+        file_metadata.path)
+    if compression_type != filesystem.CompressionTypes.UNCOMPRESSED:
+      logging.error("VCF file %s is compressed; disk requirement estimator "
+                    "will not be accurate.", file_metadata.path)
+    file_sizes.append((file_metadata.path, file_metadata.size_in_bytes,))
+  return file_sizes
+
+
+def _convert_variants_to_bytesize(variant):
+  # type: (vcfio.Variant) -> int
+  return coders.registry.get_coder(vcfio.Variant).estimate_size(variant)
+
+
+class FileSizeInfo(object):
+  def __init__(self, name, raw_file_size, encoded_file_size=None):
+    # type: (str, int, int) -> None
+    self.name = name
```

Comment:
Did you use the name anywhere?

Reply:
I use this in the `generate_report` module as well; WDYT?

Reply:
Sorry this was my fault; I either cited some earlier revision incorrectly or misspoke/misread entirely

The `self.name` is only consumed when there's some issue reading the file in `estimate_encoded_file_size` so that a more descriptive message can be logged:
```
      logging.warning("File %s appears to have no valid Variant lines. File "
                      "will be ignored for size estimation.", self.name)
```